networkit/cpp/graph/Graph.cpp

```diff
@@ -828,41 +828,42 @@ void Graph::removeEdgesFromIsolatedSet(const std::vector<node> &nodesInSet) {
 }
 
 void Graph::removeSelfLoops() {
-	count nEdgesRemoved = 0;
-
-#pragma omp parallel for reduction(+ : nEdgesRemoved)
+#pragma omp parallel for
 	for (omp_index i = 0; i < static_cast<omp_index>(z); ++i) {
 		node u = static_cast<node>(i);
 		auto isSelfLoop = [&u](node v) { return u == v; };
-		nEdgesRemoved += removeAdjacentEdges(u, isSelfLoop);
-		if (isDirected())
-			nEdgesRemoved += removeAdjacentEdges(u, isSelfLoop, true);
+		removeAdjacentEdges(u, isSelfLoop);
+		if (isDirected()) {
+			removeAdjacentEdges(u, isSelfLoop, true);
+		}
 	}
 
-	assert(nEdgesRemoved == numberOfSelfLoops());
-	m -= nEdgesRemoved;
+	m -= storedNumberOfSelfLoops;
 	storedNumberOfSelfLoops = 0;
 }
 
 void Graph::removeMultiEdges() {
-	count nEdgesRemoved = 0;
+	count removedEdges = 0;
+	count removedSelfLoops = 0;
 	std::unordered_set<node> nodes;
 
 	forNodes([&](node u) {
 		auto isMultiedge = [&nodes](node v) { return !nodes.insert(v).second; };
```

Comment:
add `nodes.reserve(degree(u))` ? not sure whether this helps

Reply:
Probably it is a bit more efficient, I'll do that.